torch/nn/modules/linear.py

```diff
@@ -178,4 +179,62 @@ def extra_repr(self) -> str:
             self.in1_features, self.in2_features, self.out_features, self.bias is not None
         )
 
+
+class LazyLinear(LazyModuleMixin, Linear):
+    r"""A :class:`torch.nn.Linear` module with lazy initialization.
+
+    In this module, the `weight` and `bias` are of :class:`torch.nn.Uninitialized`
+    class. They will be initialized  after the first call to ``forward`` is done.
+    The parameters are materialized in the `initialize_parameters` method, which
+    will instantiate them with any device or datatype the module is set to.
+    After the dry run, the module will become a :class:`torch.nn.Linear`
+    dropping all the lazy initialization logic.
+
+    Be sure to run "dummy" ``forward`` once to initialize all parameters in case
+    that you need to access the weights before the actual "real" ``forward`` pass.
+
+    Note that lazy modules cannot validate if the shape is correct during
```

Comment:
This paragraph can be removed because it's covered by the lazy module documentation above.